activerecord/lib/active_record/persistence.rb

```diff
@@ -55,6 +57,201 @@ def create!(attributes = nil, &block)
+      #   Book.insert_all([
+      #     { id: 1, title: 'Rework', author: 'David' },
+      #     { id: 1, title: 'Eloquent Ruby', author: 'Russ' }
+      #   ])
+      #
+      def insert_all(attributes, returning: nil, unique_by: nil)
+        InsertAll.new(self, attributes, on_duplicate: :skip, returning: returning, unique_by: unique_by).execute
+      end
+
+      # Inserts a single record into the databases. This method constructs a single SQL INSERT
+      # statement and sends it straight to the database. It does not instantiate the involved
+      # models and it does not trigger Active Record callbacks or validations. However, values
+      # passed to #insert! will still go through Active Record's normal type casting and
+      # serialization.
+      #
+      # See <tt>ActiveRecord::Persistence#insert_all!</tt> for documentation.
+      def insert!(attributes, returning: nil)
+        insert_all!([ attributes ], returning: returning)
+      end
+
+      # Inserts multiple records into the database. This method constructs a single SQL INSERT
+      # statement and sends it straight to the database. It does not instantiate the involved
+      # models and it does not trigger Active Record callbacks or validations. However, values
+      # passed to #insert_all! will still go through Active Record's normal type casting and
+      # serialization.
+      #
+      # The +attributes+ parameter is an Array of Hashes. These Hashes describe the
+      # attributes on the objects that are to be created. All of the Hashes must have
+      # same keys.
+      #
+      # #insert_all! will raise <tt>ActiveRecord::RecordNotUnique</tt> if any of the records being
+      # inserts would violate a unique constraint on the table. In that case, no records
+      # would be inserted.
+      #
+      # To skip duplicate records, see <tt>ActiveRecord::Persistence#insert_all</tt>.
+      # To replace them, see <tt>ActiveRecord::Persistence#upsert_all</tt>.
+      #
+      # Returns an <tt>ActiveRecord::Result</tt>. The contents of the result depend on the
+      # value of <tt>:returning</tt> (see below).
+      #
+      # ==== Options
+      #
+      # [:returning]
+      #   (Postgres-only) An array of attributes that should be returned for all successfully
+      #   inserted records. For databases that support <tt>INSERT ... RETURNING</tt>, this will default
+      #   to returning the primary keys of the successfully inserted records. Pass
+      #   <tt>returning: %w[ id name ]</tt> to return the id and name of every successfully inserted
+      #   record or pass <tt>returning: false</tt> to omit the clause.
+      #
+      # ==== Examples
+      #
+      #   # Insert multiple records
+      #   Book.insert_all!([
+      #     { title: 'Rework', author: 'David' },
+      #     { title: 'Eloquent Ruby', author: 'Russ' }
+      #   ])
+      #
+      #   # raises ActiveRecord::RecordNotUnique beacuse 'Eloquent Ruby'
```

Comment:
beacuse

Reply:
@vitalyp : It is fixed on master branch -> https://github.com/rails/rails/blob/master/activerecord/lib/active_record/persistence.rb#L171